src/hypercube.c

```diff
@@ -179,7 +180,8 @@ ts_hypercube_from_constraints(ChunkConstraints *constraints, MemoryContext mctx)
 
 			Assert(hc->num_slices < constraints->num_dimension_constraints);
 			slice = ts_dimension_slice_scan_by_id(cc->fd.dimension_slice_id, mctx);
-			Assert(slice != NULL);
+			AssertOr(ERROR, slice != NULL);
+			AssertOr(ERROR, hc->num_slices < hc->capacity);
```

Comment:
This error seems pointless/impossible given that we allocate the hypercube according to the number of dimensional constraints and then add only as many as we allocated space for.